src/MyEventEmitter.js

```diff
@@ -1,14 +1,72 @@
+/* eslint-disable max-len */
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.events = {};
+  }
+
+  checkEventName = (eventName) => {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+  };
+
+  on(eventName, listener) {
+    this.checkEventName(eventName);
+    this.events[eventName].push(listener);
+  }
+
+  once(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.on(eventName, onceListener);
+  }
+
+  off(eventName, listener) {
+    if (this.events[eventName]) {
+      this.events[eventName] = this.events[eventName].filter(
+        (registeredListener) => registeredListener !== listener,
+      );
+    }
+  }
+
+  emit(eventName, ...args) {
+    if (this.events[eventName]) {
+      this.events[eventName].forEach((listener) => {
+        listener(...args);
+      });
+    }
+  }
+
+  prependListener(eventName, listener) {
+    this.checkEventName(eventName);
+    this.events[eventName].unshift(listener);
+  }
+
+  prependOnceListener(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.prependListener(eventName, onceListener);
+  }
```

Comment:
Same issue as with the `once` method. The listener is removed before it has been invoked. Consider invoking the listener before removing it.